### ISSUE TITLE
OasisLMF install fails on OSX Catalina because of ktools installation

### ISSUE THREAD

**Issue by shaystrong** [OasisLMF/OasisLMF#884]
**IMPORTANT: Please apply the relevant labels, for example if this issue is needed as a backported fix add the label `LTS fix` (Long term support fix)

## Issue Description
MAC OSX compiling oasislmf from source or via pip install fails because of ktools install. I can install ktools separately using `./configure --enable-osx` but I can't get the oasislmf install to pick up that version. It defaults to installing it new without the `--enable-osx`.

## Steps to Reproduce (Bugs only)

1. clone the repo and ` pip install .`
OR
2. `pip install oasislmf`

## Version / Environment information
mac osx Catalina 10.15.7

## Example data / logs
```    
    depbase=`echo gulcalc.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
    	c++ -DHAVE_CONFIG_H -I. -I../..   -I/include  -std=c++11 -Wno-unused-result -fthreadsafe-statics -frtti -fno-omit-frame-pointer  -O2 -MT gulcalc.o -MD -MP -MF $depbase.Tpo -c -o gulcalc.o gulcalc.cpp &&\
    	mv -f $depbase.Tpo $depbase.Po
    In file included from gulcalc.cpp:39:
    In file included from ./gulcalc.h:41:
    ./../include/oasis.h:390:47: error: use of undeclared identifier 'getpid'
            std::string b2 = base + ":" + std::to_string(GETPID());
                                                         ^
    ./../include/oasis.h:63:16: note: expanded from macro 'GETPID'
    #define GETPID getpid
                   ^
    ./../include/oasis.h:391:46: error: use of undeclared identifier 'getpid'
            base = "log/" + base + "_" + std::to_string(GETPID()) + ".log";
                                                        ^
    ./../include/oasis.h:63:16: note: expanded from macro 'GETPID'
    #define GETPID getpid
                   ^
    2 errors generated.
    make[3]: *** [gulcalc.o] Error 1
    make[2]: *** [all-recursive] Error 1
    make[1]: *** [all-recursive] Error 1
    make: *** [all] Error 2
    Ktools build failed.
    
```

**Comment by rouvenoberholzer** [comment]
I encountered the same issue. Any ideas by when this will be fixed?

**Comment by calvinccs** [comment]
Hi there, which version of oasislmf should I install? I am using Big Sur. I notice that the mac_build branch is removed, and I have that Ktools build failed error.

**Comment by sambles** [comment]
Hi @calvinccs  the fix from that branch was merged into the main release from version `1.22.0` and up, installing the latest version _should_ work `pip install oasislmf==1.26.3`

If its failing with that package version then we might need to reopen this issue. 

#### Helpful info for debugging 
1. What python version is installed on your system? 
```
$ python --version
Python 3.9.14
```

2. What does the python detect your OS as?   If the following command returns something other than `Darwin` then that fix won't kick in. 
```
$ python -c "import platform; print(platform.system())" 
Darwin
```

**Comment by calvinccs** [comment]
I am using python 3.9.7 and python detect my OS as Darwin. However, I cannot install oasislmf==1.26.3

By the way, I have anaconda in my system, would that be an issue?
--------------
  ERROR: Command errored out with exit status 1:
   command: /Users/XXXX/opt/anaconda3/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/zc/zlcjf8r95t3fk09mg52bfzkm0000gp/T/pip-install-820vhf5t/oasislmf_28b5c6cf48dd4c039af08091c95ba0c1/setup.py'"'"'; __file__='"'"'/private/var/folders/zc/zlcjf8r95t3fk09mg52bfzkm0000gp/T/pip-install-820vhf5t/oasislmf_28b5c6cf48dd4c039af08091c95ba0c1/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/zc/zlcjf8r95t3fk09mg52bfzkm0000gp/T/pip-wheel-yhahbzs7
       cwd: /private/var/folders/zc/zlcjf8r95t3fk09mg52bfzkm0000gp/T/pip-install-820vhf5t/oasislmf_28b5c6cf48dd4c039af08091c95ba0c1/
  Complete output (205 lines):
  /Users/XXXX/opt/anaconda3/lib/python3.9/site-packages/setuptools/dist.py:717: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead
    warnings.warn(
  running bdist_wheel
  The [wheel] section is deprecated. Use [bdist_wheel] instead.
  running build
  running build_py
  creating build
......
......
......
......
Ktools build failed.
